recipe/meta.yaml

```diff
@@ -11,10 +11,12 @@ source:
     folder: rmm
 
 build:
-  number: 1
+  number: 2
   skip: true  # [(not linux) or (cuda_compiler_version == "None")]
   run_exports:
     - {{ pin_subpackage(name, max_pin="x.x") }}
+    - {{ pin_compatible("thrust", max_pin="x.x") }}
+    - {{ pin_compatible("cub", max_pin="x.x") }}
```

Comment:
Do we actually need `run_exports` here? A downstream project could 100% be using rmm purely as a build dependency and not leaking it into any of its headers for example.

Reply:
This is the error I was trying to fix:
```
2022-05-25T06:33:21.7039933Z   /home/conda/feedstock_root/build_artifacts/rmm_1653459740306/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include/thrust/system/cuda/config.h:79:2: error: #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.
2022-05-25T06:33:21.7042122Z      79 | #error The version of CUB in your include path is not compatible with this release of Thrust. CUB is now included in the CUDA Toolkit, so you no longer need to use your own checkout of CUB. Define THRUST_IGNORE_CUB_VERSION_CHECK to ignore this.
2022-05-25T06:33:21.7043060Z         |  ^~~~~
```
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=511495&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=672

It happened there because librmm did not take care of the dependencies, so Thrust/CUB had to become an explicit dependency for downstream, which is counter-intuitive. Ideally they should be hidden by librmm, shouldn't they?

Reply:
No, RMM being header only means that `thrust` and `cub` are needed as run dependencies so that if someone has `librmm` in their `host` section it additionally gets `thrust` and `cub` in that `host` section.

It doesn't need the `run_exports` though as that would add `librmm` to the `run` section of that downstream recipe, potentially unnecessarily.

Reply:
Fixed in a0ab5ee.